server/src/config/mod.rs

```diff
@@ -23,6 +23,10 @@ pub struct ServerConfig {
     pub artifact_stage: String,
     /// What branch should be downloaded
     pub target_branch: String,
+    /// What variable should be expected to be present in the Pipeline Webhook from gitlab.
+    /// NOTE: an empty `target_variable` will pass all artifacts
+    /// NOTE: if both `target_branch` and `target_variable` is present, both filters will be applied
+    pub target_variable: String,
```

Comment:
probably good to make it an Option which clarifies and encodes the intent into the typesystem.

I would rephrase the comment to:
```
/// If set only pipeline events with this variable will be processed. Otherwise every event will be considered.
```